lib/AST/Type.cpp

```diff
@@ -3192,7 +3192,8 @@ Type TypeBase::getSuperclassForDecl(const ClassDecl *baseClass,
 
     t = t->getSuperclass(useArchetypes);
   }
-  llvm_unreachable("no inheritance relationship between given classes");
+
+  return ErrorType::get(this);
```

Comment:
How did we end up here? Name lookup shouldn't be finding members of unrelated classes...

Reply:
I suspect that Decl-based name lookup is finding something via a superclass (decl), but the types are invalid so we don't see them in the superclass (Type) chain. I can try to narrow this.

Reply:
See https://github.com/apple/swift/pull/18720/commits/50763fc641a29cb178fa7e6f60ac6c81bb77fad5